synthtool/gcp/common.py

```diff
@@ -57,6 +57,11 @@ def _generic_library(self, directory: str, **kwargs) -> Path:
                 self.excludes.append("samples/README.md")
 
         t = templates.TemplateGroup(self._template_root / directory, self.excludes)
+
+        # TODO: migrate to python.py once old sample gen is deprecated
+        if directory == "python_samples":
```

Comment:
I'd like to keep language-specific code out of _generic_library().  Is there another place we could put this code?

Reply:
I had the same concern, unfortunately it needs to be added to `t` which is only initialized on the line above in this method. We could not call generic_library and inline the code, but that could be harder to maintain. WDYT?

Reply:
Would you be able to add some sort of `generate_sample_help` tag into `**kwargs` to dictate this if branch instead of `directory == "python_samples"`?
And then pass in `generate_sample_help: True` thru the kwargs in when calling `_generic_library()` from `py_samples`?

Reply:
That makes sense, thank you! We could do that, and delegate the language specific types of help output in the `_get_help` method (ie right now it only works for python, but if other languages wanted a similar feature, we could handle what `_get_help` returns based on the language calling it. @SurferJeffAtGoogle thoughts?

Reply:
I spent about 25 minutes studying common.py from top to bottom, and it's a heap of technical debt in need of a refactor that is way outside the scope of this PR, so this code is fine.